testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -431,48 +431,59 @@ protected SqlOperatorFixture fixture() {
         true);
   }
 
-  /** Tests that CAST and SAFE_CAST are basically equivalent but SAFE_CAST is
-   * only available in BigQuery library. */
-  @Test void testCastVsSafeCast() {
```

Comment:
adding a dialect udf shouldn't have touched so many existing tests, can you explain why?

Reply:
Because I want to reuse most of the test code for `CAST` and `SAFE_CAST`, just like what we did in
https://github.com/apache/calcite/pull/3093

Reply:
in my understanding parameterized tests are better suited for this.
I'd suggest some refactoring into these tests. let's see what others think

Reply:
In this case the behavior of the functions (`SAFE_CAST`, `TRY_CAST`) are identical, it is just named differently across dialects, so I am a fan of the code reuse.

Reply:
@zinking All the tests in this class are in form of parameterized test except `testCastVsSafeCastVsTryCast`, such as `testCastToString`, do you mean I also change this test into a parameterized one?

Reply:
right, that would be better in my unserstanding

Reply:
@zinking I have modified this, please have another look when you have time